src/components/chipsInput/index.js

```diff
@@ -96,27 +96,32 @@ class ChipsInput extends Component {
       value: props.value,
       tags: _.cloneDeep(props.tags) || [],
       tagIndexToRemove: undefined,
-      initialTags: props.tags
+      initialTags: props.tags,
+      inputRef: undefined
     };
   }
 
   componentDidMount() {
     if (Constants.isAndroid) {
-      const textInputHandle = ReactNative.findNodeHandle(this.input);
+      const textInputHandle = ReactNative.findNodeHandle(this.state.inputRef);
       if (textInputHandle && NativeModules.TextInputDelKeyHandler) {
         NativeModules.TextInputDelKeyHandler.register(textInputHandle);
       }
     }
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
+    const nextState = {}
     if (nextProps.tags !== prevState.initialTags) {
-      return {
-        initialTags: nextProps.tags,
-        tags: nextProps.tags
-      };
+      nextState.initialTags = nextProps.tags;
+      nextState.tags = nextProps.tags;
     }
-    return null;
+    if (nextProps.tags.length > prevState.initialTags.length) {
```

Comment:
What if tags.length is smaller than the initialTags's?

Reply:
I guess the text should still be there. It means that the user just removed a tag.
In out scenario we show a list based on the text, so there is not reason to reset it. I think that having a new tag is the only case

Reply:
@Inbal-Tish I edited the description.
Please review the new commit